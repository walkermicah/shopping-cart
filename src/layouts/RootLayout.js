import { Outlet, Link, NavLink } from 'react-router-dom';

import logo from '../img/logo.svg';

import '../styles/layouts/RootLayout.scss';

function RootLayout({ nItems }) {
  return (
    <div className="RootLayout">
      <header>
        <nav className="RootLayout-nav">
          <Link to="/" className="RootLayout-nav-title">
            <h1>InstaCarrot</h1>
            <img src={logo} alt="logo"></img>
          </Link>

          <ul>
            <li>
              <NavLink to="/">Home</NavLink>
            </li>
            <li>
              <NavLink to="products">Products</NavLink>
            </li>
            <li>
              <NavLink to="contact">Contact</NavLink>
            </li>
          </ul>

          <Link to="cart" className="RootLayout-nav-link">
            <i className="fa-solid fa-cart-shopping"></i>
            <span>{nItems}</span>
          </Link>
        </nav>
      </header>

      <main>
        <Outlet />
      </main>
    </div>
  );
}

export default RootLayout;
