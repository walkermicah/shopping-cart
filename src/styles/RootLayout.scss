.RootLayout {
  height: 100vh;
  font-family: 'Lato', sans-serif;
  font-size: 2rem;
  font-weight: 300;

  &-nav {
    display: flex;
    align-items: center;
    padding: 0.5rem 1rem;
    background-color: rgb(0, 0, 0);
    color: rgb(255, 255, 255);

    &-title {
      display: flex;
      align-items: center;

      h1 {
        font-family: 'Itim', cursive;
        font-size: 5rem;
      }

      img {
        width: 6rem;
        height: 6rem;
        transform: translateX(-1rem) rotate(45deg);
      }
    }

    ul {
      display: flex;
      gap: 2rem;
      list-style: none;
      margin-left: auto;
    }

    li,
    &-link {
      transition: transform 0.2s;
    }

    li:hover,
    &-link:hover {
      transform: translateY(-1px);
    }

    a {
      color: rgb(255, 255, 255);
      text-decoration: none;
      padding: 1rem;
    }

    a.active {
      font-weight: 700;
    }

    &-link {
      margin: 2rem;
      position: relative;

      i {
        font-size: 2.5rem;
      }

      span {
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: rgb(255, 135, 25);
        color: black;
        height: 2rem;
        width: 2rem;
        border-radius: 50%;
        font-size: 1rem;
        font-weight: 700;
        position: absolute;
        top: 0%;
        left: 70%;
      }
    }
  }
}

/* 600px  */
@media only screen and (max-width: 37.5em) {
  .RootLayout-nav {
    display: grid;
    grid-template-areas:
      'title cart'
      'links links';

    ul {
      grid-area: links;
      margin-left: 0;
      justify-content: center;
      padding: 0.5rem 0rem;
    }

    &-link {
      margin-left: auto;
    }
  }
}
