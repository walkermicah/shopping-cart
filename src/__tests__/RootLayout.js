import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import RootLayout from '../layouts/RootLayout';

describe('Root Layout Component', () => {
  it('Renders root layout header and main', () => {
    const { container } = render(<RootLayout />, { wrapper: BrowserRouter });
    expect(container).toMatchSnapshot();
  });

  it('Displays number of items in cart', () => {
    render(<RootLayout nItems={5} />, { wrapper: BrowserRouter });
    const span = screen.getByTestId('n-items');
    expect(+span.textContent).toBe(5);
  });
});
