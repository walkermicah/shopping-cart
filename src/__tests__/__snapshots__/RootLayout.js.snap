// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Layout Component Renders root layout header and main 1`] = `
<div>
  <div
    class="RootLayout"
  >
    <header>
      <nav
        class="RootLayout-nav"
      >
        <a
          class="RootLayout-nav-title"
          href="/"
        >
          <h1>
            InstaCarrot
          </h1>
          <img
            alt="logo"
            src="logo.svg"
          />
        </a>
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class=""
              href="/products"
            >
              Products
            </a>
          </li>
          <li>
            <a
              class=""
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
        <a
          class="RootLayout-nav-link"
          href="/cart"
        >
          <i
            class="fa-solid fa-cart-shopping"
          />
          <span
            data-testid="n-items"
          />
        </a>
      </nav>
    </header>
    <main />
  </div>
</div>
`;
